scuba/config.py

```diff
@@ -399,9 +412,14 @@ def _expand_path(in_str: str, base_dir: Optional[Path] = None) -> Path:
 @dataclasses.dataclass(frozen=True)
 class ScubaVolume:
     container_path: Path
-    host_path: Path  # TODO: Optional for anonymous volume
+    host_path: Optional[Path] = None
+    volume_name: Optional[str] = None
     options: List[str] = dataclasses.field(default_factory=list)
 
+    def __post_init__(self) -> None:
+        if sum(bool(x) for x in (self.host_path, self.volume_name)) != 1:
+            raise ValueError("Exactly one of host_path, volume_name must be set")
```

Comment:
I really like the clarity of this error message. However, it provides a hint to issues with `yaml` de-serialization. See below for an example with multiple `name` keys defined for a single complex `volume`. All `yaml` config would really benefit from a defined schema with constraint validation.

I'm not suggesting you need to deal with this case, but it's annoying that simple typos can fail with success.

```yml
volumes:
  /go/pkg/mod/cache:
      name: go-mod-cache
      name: go-mod-caches
```
```javascript
// docker inspect 02e871026589 | jq '.[0].Mounts | map(select(.Type=="volume"))'
[
  {
    "Type": "volume",
    "Name": "go-mod-caches",
    "Source": "/var/lib/docker/volumes/go-mod-caches/_data",
    "Destination": "/go/pkg/mod/cache",
    "Driver": "local",
    "Mode": "z",
    "RW": true,
    "Propagation": ""
  }
]
```